### PR TITLE
fix: failed to create blanco database after selecting a different template (+ related fixes)

### PR DESCRIPTION
**Pull request by svandenhoek**
What are the main changes you did:

Using http://localhost:8080/apps/molgenis-components/#/component/InputSelect & http://localhost:8080/apps/central/#/ to explain changes.

### Old behavior
1) `required` showed the first element was automatically selected, yet no `modelvalue` was set and the first element could only be selected by first selecting another option and then selecting the first one again.

https://github.com/molgenis/molgenis-emx2/assets/7658246/f8f52384-8241-4631-8157-2d11a20107f6

2) If no `required` & `v-model=null`, `modelvalue` is not set. However, if selecting another option and the selecting the empty option again, it becomes `modelvalue=""` instead. 

https://github.com/molgenis/molgenis-emx2/assets/7658246/87aaf9bb-8c11-4f06-a043-9d8fcea3ac96

This caused issues in f.e. Database creation with no template:

https://github.com/molgenis/molgenis-emx2/assets/7658246/795c5491-3628-4a94-b7c2-783196cef2d3

Scherm­afbeelding 2024-06-27 om 18 49 52
Scherm­afbeelding 2024-06-27 om 18 50 18

3) There was a duplicate empty value when creating a new Database.
Scherm­afbeelding 2024-06-27 om 19 04 04

### New behavior
1) If `required`, an `$emit("update:modelValue"`  is triggered at `<select :modelValue=` to update it to the first item in `options` instead (simply returning the new value without an update showed the outdated value still in HTML).

2) If a `@change` is detected with a `$event.target.value ==''`, `$emit("update:modelValue", null)` is done instead of using the actual value. This causes the `modelvalue` attribute to be removed from `select`, similar to if no value was selected yet.

https://github.com/molgenis/molgenis-emx2/assets/7658246/8c83380d-ac16-4e58-bd4f-2af2a80fd946

3) Removed the `null` from `options` when creating a new Database, as a placeholder option is already generated when not using `required`.

how to test:
- Go to http://localhost:8080/apps/molgenis-components/#/component/InputSelect
- Validate whether the value shown after `Selected: ` is equal to what is actually selected in the drop-down menu.
- Validate whether the value shown after `Selected: ` is equal to the `modelvalue` as shown in the HTML (browser developer tools).
- Check using browser developer tools whether `modelvalue` attribute is not present if the blank value is selected, even after selecting other values.
- When creating a new database with blank template, it should still work even after having selected a different template.

todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/InputSelect.vue [hunk 1]
```diff
@@ -12,10 +12,10 @@
...
       v-else-if="!readonly"
       :id="id"
-      :modelValue="modelValue"
+      :modelValue="defaultModelValue(modelValue, required, options)"
       :readonly="readonly"
       class="form-control"
-      @change="$emit('update:modelValue', $event.target.value)"
+      @change="updateModelValue($event.target.value)"
     >
       <option
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/InputSelect.vue:20]: I think a better solution would be to add an empty option?

**Review by mswertz — CHANGES_REQUESTED** [review]
please ask a javascript programmer. I think normally an empty option would be the recommended solution so you can also set to 'undefined'.

File: apps/molgenis-components/src/components/forms/InputSelect.vue (hunk 1 above)
Comment (svandenhoek) [reply apps/molgenis-components/src/components/forms/InputSelect.vue:20]: Indeed. The issue here is that the behaviour between "never clicked on any other option" vs "clicked a different option and then selected the empty option again" was different.

File: apps/molgenis-components/src/components/forms/InputSelect.vue [hunk 2]
```diff
@@ -57,6 +57,19 @@ export default {
   props: {
     options: { type: Array, required: true },
   },
+  methods: {
+    defaultModelValue: function (modelValue, required, options) {
+      if ((modelValue === undefined || modelValue === null) && required) {
+        this.updateModelValue(options[0]);
```
[the remark this answers is not in the document]
Comment (svandenhoek) [reply apps/molgenis-components/src/components/forms/InputSelect.vue:63]: That or we should simply state if you use `required`, one should select a value and NOT USE `null`/`undefined`.

[the remark this answers is not in the document]
Comment (svandenhoek) [reply apps/molgenis-components/src/components/forms/InputSelect.vue:63]: Oh, **to explain the initial issue:** Reason for this fix was that if `required`, the drop-down menu showed as-if the first element was already selected while in reality it was not.

**Review by mswertz — APPROVED** [review]

**Pull request merged**